cpp/environment/MatchEnv.cc

```diff
@@ -228,10 +235,10 @@ std::vector<vec3<float>> EnvDisjointSet::getAvgEnv(const unsigned int m)
 
     // loop through the vectors in env now, dividing by the total number
     // of contributing particle environments to make an average
-    for (unsigned int n = 0; n < m_max_num_neigh; n++)
+    for (auto& vec : env)
     {
-        vec3<float> normed = env[n] / static_cast<float>(N);
-        env[n] = normed;
+        vec3<float> normed = vec / static_cast<float>(N);
+        vec = normed;
```

Comment:
Is there a reason not to do this
```suggestion
        vec = vec / static_cast<float>(N);
```